cmd/server-main.go

```diff
@@ -277,7 +281,93 @@ func mergeServerCtxtFromConfigFile(configFile string, ctxt *serverCtxt) error {
 		ctxt.SFTP = append(ctxt.SFTP, fmt.Sprintf("ssh-private-key=%s", cf.Options.SFTP.SSHPrivateKey))
 	}
 
-	ctxt.Layout, err = buildDisksLayoutFromConfFile(cf.Pools)
+	ctxt.Layout, err = buildDisksLayoutFromConfFile(cf.EnableExpandPools, cf.Pools, false)
+
+	if ctxt.EnableExpandPools {
+		stat, err := os.Stat(configFile)
+		if err != nil {
+			return err
+		}
+		ltime := stat.ModTime()
+		go func() {
+			ticker := time.NewTicker(time.Second * 10)
+			for {
+				select {
+				case <-ticker.C:
+					lstat, err := os.Stat(configFile)
+					if err == nil {
+						if lstat.ModTime().After(ltime) {
+							rd, err := Open(configFile)
+							if err == nil {
+								defer rd.Close()
+								cf := &config.ServerConfig{}
+								dec := yaml.NewDecoder(rd)
+								dec.SetStrict(true)
+								if err = dec.Decode(cf); err == nil {
+									globalServerCtxt.poolsRWMutex.Lock()
```

Comment:
Why this lock? Are you expecting this to be called concurrently? As far as I can tell this is only called once?